### ISSUE TITLE
Slot Prop Types

### ISSUE THREAD

**Issue by JohnCampionJr** [johnsoncodehk/volar#67]
Is there any guidance on properly coding slot properties for type inference?

Getting error `Property 'default' does not exist on type '{}'.ts(2339)` on a very simple slot. Error is on this line
```html
<Authorized v-slot="{ userName }">
{{ userName }}
</Authorized>
```

Authorized.vue is this

```ts
<template>
  <template v-if="isAuthorized">
    <slot :userName="userName"></slot>
  </template>
</template>

<script setup lang="ts">
import { computed } from 'vue'
import authStore from '~/store/authStore'

const userName = computed(() => authStore.userName)
const isAuthorized = computed(() => authStore.isAuthenticated)
</script>
```

userName is typed as a string in authStore.  It is converted to a `ComputedRef<string>`

I don't know how to remove the error message.

**Comment by johnsoncodehk** [comment]
it's seem a bug and I have note it: https://github.com/johnsoncodehk/volar/projects/4#card-55169086

wait a minute I'm checking.

**Comment by johnsoncodehk** [comment]
fixed in 0.21.11.

**Comment by JohnCampionJr** [comment]
Confirmed; thanks so much!
